File: 백준/Silver/1436. 영화감독 숌/영화감독 숌.cc
#define _CRT_SECURE_NO_WARNINGS
#include <stdio.h>
#include <string.h>
#include <math.h>

int main(void)
{
	int n;
	int index = 0;
	int num = 665;

	scanf("%d", &n);

	while (1)
	{
		if (index == n)
			break;
		num++;

		for (int i = 0; i < 10; i++)
			if (num / (int)pow(10, i) % 1000 == 666)
			{
				index++;
				break;
			}
	}
	printf("%d", num);

	return 0;
}

File: 백준/Silver/1436. 영화감독 숌/영화감독 숌.java
import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStreamReader;
import java.util.*;

public class Main {
	public static void main(String args[]) throws Exception {
		BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
		StringTokenizer st = new StringTokenizer(br.readLine());

		int n = Integer.parseInt(st.nextToken());
		
		int idx = 0;
		int num = 0;
		while(true) {
			if(idx == n)
				break;
			num++;
			
			for(int i=0;i<10;i++) {
				if(num/ (int)Math.pow(10, i) % 1000 == 666) {
					idx++;
					break;
				}
			}
		}

		System.out.println(num);
	}
}